runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -215,13 +215,25 @@ public void onGameTick(final GameTick event)
 		{
 			int objectId = ((ObjectClueScroll) clue).getObjectId();
 
+			final WorldPoint location = ((LocationClueScroll) clue).getLocation();
+
 			if (objectId != -1)
 			{
 				GameObjectQuery query = new GameObjectQuery().idEquals(objectId);
 				objectsToMark = queryRunner.runQuery(query);
 
 				// Set hint arrow to first object found as there can only be 1 hint arrow
-				if (objectsToMark.length >= 1)
+				if (location != null)
+				{
+					for (GameObject object : objectsToMark)
+					{
+							if (object.getWorldLocation() == location)
+							{
+								client.setHintArrow(object.getWorldLocation());
```

Comment:
Break out the loop after this since you have found your object.

Reply:
I was literally pushing that when you commented

Reply:
Why find a match at this point then? Same effect would be achieved with checking if location != null and not adding any object marker at that point, as location is set before. Also, there is no guarantee that ObjectClueScroll will be also LocationClueScroll and you are casting it to LocationClueScroll without doing any checks here.